glib2/lib/glib2/variant.rb

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2015-2022  Ruby-GNOME2 Project Team
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+module GLib
+  class Variant
+    class << self
+      alias_method :parse_raw, :parse
+
+      def parse(string, type = nil)
+        parse_raw(string, type)
+      end
```

Comment:
I'm not very familiar with the C API. I may be able to guess out how to implement an optional arg in C, from some existing example in the API.

Albeit from a limited perspective about available approaches, I'd thought it might be simpler to use a constant args list in C and mange the fully Ruby args syntax in Ruby?